runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -695,13 +683,17 @@ impl<T: Config> Pallet<T> {
 					false,
 				);
 
-			let freed = collect_all_freed_cores::<T, _>(freed_concluded.iter().cloned());
-
-			<scheduler::Pallet<T>>::clear();
+			let mut freed = collect_all_freed_cores::<T, _>(freed_concluded.iter().cloned());
```

Comment:
Maybe `collect_all_freed_cores` should call `collect_disputed`? After all it is not all freed cores.

Reply:
If you want to make a separate PR against master, and merge master afterwards - agreed that would be even better!